code/game/objects/items/devices/rigbuilding.dm

```diff
@@ -0,0 +1,145 @@
+/obj/item/device/rigframe
+	name = "Hardsuit frame"
+	desc = "An incomplete hardsuit frame, lacking all sealing and armor plating. Its unfinished state makes it impossible to actually wear."
+	icon_state = "rigframe_0"
+	item_state = null
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/spacesuits.dmi', "right_hand" = 'icons/mob/in-hand/right/spacesuits.dmi')
+	w_class = W_CLASS_LARGE
+	var/buildstate = 0
+	var/obj/item/weapon/cell/cell = null
+	var/obj/item/clothing/suit/space/rig/result = null //changes based on which kit you use
+
+/obj/item/device/rigframe/update_icon()
+	if(buildstate == 0)
+		icon_state = "rigframe_0"
+	else if(buildstate == "wired")
+		icon_state = "rigframe_1"
+		desc = "An incomplete hardsuit frame, lacking all sealing and armor plating. Its unfinished state makes it impossible to actually wear. The wiring is installed, but not secure."
+	else if(buildstate == "wired_secure")
+		icon_state = "rigframe_2"
+		desc = "An incomplete hardsuit frame, lacking all sealing and armor plating. Its unfinished state makes it impossible to actually wear. Though the wiring is secure, it lacks a power source."
+	else if(buildstate == "cell")
+		icon_state = "rigframe_2"
+		desc = "An incomplete hardsuit frame, lacking all sealing and armor plating. Its unfinished state makes it impossible to actually wear. This frame is ready to recieve its plating."
+	else if(buildstate == "plate")
+		icon_state = "rigframe_3"
+		desc = "A nearly-complete hardsuit frame. The plates are installed, but not yet fastened. You couldn't wear it without it falling apart around you."
+	else
+		icon_state = "rigframe_0" // shouldn't happen
+
+/obj/item/device/rigframe/Destroy()
+	qdel(cell)
+	cell = null
+	..()
+
+/obj/item/device/rigframe/attackby(obj/item/weapon/W, mob/user)
+	if(src.loc == user)
+		to_chat(user, "You need to place \the [src] on the ground before modifying it.")
+		return
+
+	else if(istype(W, /obj/item/stack/cable_coil))
+		var/obj/item/stack/cable_coil/C = W
+		if(buildstate == 0)
+			if(C.amount < 5)
```

Comment:
you can reduce linecount here with

```
if(C.use(5))
   //success
else
  //fail message
```

instead of having use and the amount check be separate